mongo/collection.go

```diff
@@ -293,7 +293,22 @@ func (coll *Collection) insert(ctx context.Context, documents []interface{},
 	}
 	op = op.Retry(retry)
 
-	return result, op.Execute(ctx)
+	err = op.Execute(ctx)
+	wce, ok := err.(driver.WriteCommandError)
+	if !ok {
+		return result, err
+	}
+
+	for i, we := range wce.WriteErrors {
```

Comment:
Add some comments about what this is doing. Also, add a comment to explain the `idIndex := int(we.Index) - i` line. I initially didn't realize we needed it because the `result` slice is changing in the loop.